lang/java/reef-common/src/main/java/org/apache/reef/driver/catalog/EvaluatorInfo.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.driver.catalog;
+
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.annotations.audience.Private;
+
+/**
+ * Provides information on an evaluator. Currently primarily used in the case of a recovered evaluator
+ * on driver restart.
+ */
+@Private
+@DriverSide
+public final class EvaluatorInfo {
+  private final String rackName;
+  private final int numberOfCores;
+  private final int memoryInMegaBytes;
+  private final String nodeId;
+
+  public EvaluatorInfo(final String nodeId,
```

Comment:
Can you rationalize the difference between this and `org.apache.reef.driver.evaluator.EvaluatorDescriptor`? We could extend the latter with the fields not yet in it?